mesonpy/__init__.py

```diff
@@ -717,6 +717,40 @@ def __init__(
                     ''')
                     self._meson_cross_file.write_text(cross_file_data, encoding='utf-8')
                     self._meson_args['setup'].extend(('--cross-file', os.fspath(self._meson_cross_file)))
+        elif sysconfig.get_platform().startswith('ios-'):
+            ios_ver = platform.ios_ver()
+
+            arch = platform.machine()
+            family = 'aarch64' if arch == 'arm64' else arch
+            subsystem = 'ios-simulator' if ios_ver.is_simulator else 'ios'
+            min_ios_version = os.getenv('IPHONEOS_DEPLOYMENT_TARGET', ios_ver.release)
+            prefix = sysconfig.get_config_var('prefix')
+
+            cross_file_data = textwrap.dedent(f'''
+                [binaries]
+                c = ['{arch}-apple-{subsystem}-clang', '-mios-version-min={min_ios_version}']
+                cpp = ['{arch}-apple-{subsystem}-clang++', '-mios-version-min={min_ios_version}']
+                objc = ['{arch}-apple-{subsystem}-clang', '-mios-version-min={min_ios_version}']
+                objcpp = ['{arch}-apple-{subsystem}-clang++', '-mios-version-min={min_ios_version}']
+                ar = ['{arch}-apple-{subsystem}-ar']
+
+                [built-in options]
+                c_link_args = ['-mios-version-min={min_ios_version}', '-F', {prefix!r}, '-framework', 'Python']
+                cpp_link_args = ['-mios-version-min={min_ios_version}', '-F', {prefix!r}, '-framework', 'Python']
+                objc_link_args = ['-mios-version-min={min_ios_version}', '-F', {prefix!r}, '-framework', 'Python']
+
+                [host_machine]
+                system = 'ios'
+                subsystem = {subsystem!r}
+                cpu = {arch!r}
+                cpu_family = {family!r}
+                endian = 'little'
+
+                [properties]
+                longdouble_format = 'IEEE_DOUBLE_LE'
```

Comment:
This property is numpy-specific. I don't mind hardcoding that in NumPy (skip the runtime check if we're cross-compiling to iOS), but we shouldn't need to add it here.